packages/api-plugin-fulfillment-method-pickup-store/src/getFulfillmentMethodsWithQuotesPickupStore.js

```diff
@@ -0,0 +1,75 @@
+import Logger from "@reactioncommerce/logger";
+import collectStoreDetails from "./util/collectStoreDetails.js";
+
+const packageName = "fulfillment-method-pickup-store";
+const fulfillmentTypeName = "pickup";
+const fulfillmentMethodName = "store";
+const logCtx = { name: "fulfillment-method-pickup-store", file: "getFulfillmentMethodsWithQuotesPickupStore" };
+
+/**
+ * @summary Returns a list of fulfillment method quotes based on the items in a fulfillment group.
+ * @param {Object} context - Context
+ * @param {Object} commonOrder - details about the purchase a user wants to make.
+ * @param {Array} [previousQueryResults] - an array of shipping rates and
+ * info about failed calls to the APIs of some shipping methods providers
+ * @returns {Array} - an array that contains two arrays: the first array will
+ * be an updated list of shipping rates, and the second will contain info for
+ * retrying this specific package if any errors occurred while retrieving the
+ * shipping rates.
+ * @private
+ */
+export default async function getFulfillmentMethodsWithQuotesPickupStore(context, commonOrder, previousQueryResults = []) {
+  const { collections: { Fulfillment } } = context;
+  const [rates = [], retrialTargets = []] = previousQueryResults;
+  const currentMethodInfo = { packageName };
+
+  if (retrialTargets.length > 0) {
+    const isNotAmongFailedRequests = retrialTargets.every((target) => target.packageName !== packageName);
+    if (isNotAmongFailedRequests) {
+      return previousQueryResults;
+    }
+  }
+
+  const pickupDocs = await Fulfillment.find({
```

Comment:
Why are we searching for multiple fulfillment types here? Based on the `createFulfillmentType` mutation, we are allowed to have only only fulfillment entry in the database with specific fulfillment type, in this case the `pickup` type.

Reply:
I am not sure if I got your question correct.

In the `Fulfillment` collection we have one entry for each `fulfillmentType`. Each of this entry could have multiple `fulfillmentMethods`. So here, assuming we are looking for 'pickup' type, we are getting the document for 'pickup' from DB and returning the 'rates' for all the methods under 'pickup' (example 'store', 'kerbside' etc)

Reply:
That's right, but we are not fetching the fulfillment methods here, but rather the parent fulfillment type. So it's always going to be one result.

Reply:
The iteration over all fulfillment methods is happening later on line 48.

Reply:
Yes, I shall change to findOne as you suggested below